docs/ORTModule_Training_Guidelines.md

```diff
@@ -183,6 +183,16 @@ data sparsity based performance optimizations.
 	export ORTMODULE_ENABLE_EMBEDDING_SPARSE_OPTIMIZER=0 # Disable
 	```
 
+#### ORTMODULE_CACHE_DIR
+
+- **Feature Area**: *ORTMODULE/RuntimeOptions*
+- **Description**: By default, this is disabled. This env var can be used to cache the exported model for future runs.
+
+	```bash
+	export ORTMODULE_CACHE_DIR="/path/to/cache_dir" # Enable
+	unset ORTMODULE_CACHE_DIR # Disable
```

Comment:
do you want to add ORTMODULE_CACHE_PREFIX here too?

Reply:
Couple of questions:
1. Do we need multiple env vars? Can we work with just one?
2. Are both required, or CACHE_DIR is required and CACHE_PREFIX is optional. Documentation should mention this. If 1 is optional then it should also mention when the optional arg is not set. 
3. What is the criteria for picking up cached model? 
4. How does the user know if the cached model is being used or not?
5. The guideline should also include some guidance around when should this be used or why should anyone use this feature or what is the use of this feature. This will help the users understand if they should try it and also set some expectations around what should they expect from this.

Reply:
1. We do not, adjusted the code to only require ORTMODULE_CACHE_DIR
2. Not relevant after change made regarding #1
3. The filename is the hash of the model architecture gathered from `str(_flattened_module)` and if the same model architecture hash is detected, cached model will be picked up.
4. Added logs for user.
5. Added more details to guidelines.